controllers/thumbnailselection.rb

```diff
@@ -0,0 +1,62 @@
+class ::TopicPreviews::ThumbnailselectionController < ::ApplicationController
+  def index
+    params.require(:topic)
+
+    raise Discourse::InvalidAccess.new unless current_user
+
+    topic_number = params[:topic].to_i
+
+    @topic = Topic.find(topic_number)
+    @user_id = @topic.user_id
+    @posts = @topic.posts
+    @thumbnails = []
+
+    if current_user.id = @user_id || current_user.admin == true
+      @posts.map do |post|
+        @post_id = post.id
+        @doc = Nokogiri::HTML( post.cooked )
+        @img_srcs = @doc.css('img').map{ |i| i['src'] }
+        @img_srcs.map do |image|
+          if (!image.include? "emoji") && (!image.include? "avatar")
+            @thumbnails << {image: image, post_id: @post_id}
+          end
+        end
+      end
+    end
+
+    respond_to do |format|
+        format.json { render json: @thumbnails}
+    end
+  end
+
+  def update
```

Comment:
@merefield Sorry, but we have to convert this into a block running on the topic create / update hooks as I suggested in my first review. It may seem simpler to do it this way now, but it'll make it harder in the future. 

It doesn't have to work on topic create out of the box. Have a look at how the Events and Locations plugins do this. Basically you just need to change the update method action you've go here to:

```
PostRevisor.track_topic_field(:image_url) do |tc, image_url|
  @topic.update_column(:image_url, image_url)
   etc 
end
```
Then change the client-side connector to ``edit-topic`` instead of ``topic-title``, and add the selected ``image_url`` to the buffered model. This way you don't need an extra route, have to worry about route action authentication etc (which is currently incorrect), and a whole bunch of other things. 

Keep the index route, but just use the existing topic update method, rather than creating a seperate one.

Reply:
From a functional perspective I have an issue with moving it into the Topic composer.  You see the scope of the thumbnail selector is greater than the first post (because thumbnails can come from any of the posts not just the first one) so having the control at an entire post stream level makes more sense to me.  I get the technical convenience but to me the scope feels wrong?

Reply:
I agree, however the ``edit-topic`` outlet is in the edit topic UI, not in the composer. The natural place for this is in the ``edit-topic`` UI, using the existing topic edit routes.

<img width="792" alt="Screen Shot 2019-05-20 at 3 36 30 PM" src="https://user-images.githubusercontent.com/5931623/58002873-dee27b80-7b22-11e9-82bd-e25cfbdba577.png">

Reply:
Ah, got you.  Will revert

Reply:
This is now done